### ISSUE TITLE
Phase 4.1: Add CLI commands for advanced financial metrics

### ISSUE THREAD

**Issue by vibeprogrammer** [vibeprogrammer/trust#55]
## Overview
Implement advanced trading performance metrics with CLI commands for profit factor, expectancy, Sharpe ratio, and other sophisticated analytics.

### `trust metrics advanced`
Display comprehensive financial metrics:
```bash
trust metrics advanced [--account ACCOUNT_ID] [--days 90]

Output:
Advanced Trading Metrics (Last 90 days)
======================================
Returns Analysis:
├─ Total Return: +12.4% ($12,400)
├─ Annualized Return: +54.2%
├─ Risk-Free Rate: 5.0% (configurable)
└─ Excess Return: +49.2%

Risk-Adjusted Performance:
├─ Sharpe Ratio: 2.31 (Excellent)
├─ Sortino Ratio: 3.45 (Downside-focused)
├─ Calmar Ratio: 4.22 (Return/Max DD)
└─ Modified Sharpe: 2.18 (Skew-adjusted)

Trade Quality Metrics:
├─ Profit Factor: 2.85 (Gross Profit ÷ Gross Loss)
├─ Expectancy: +$425 per trade
├─ Payoff Ratio: 2.1:1 (Avg Win ÷ Avg Loss)
└─ Kelly Criterion: 18.5% (theoretical optimal sizing)

Distribution Analysis:
├─ Win Rate: 68.5% (37 wins, 17 losses)
├─ Largest Win: +$2,840 (5.2R)
├─ Largest Loss: -$890 (-1.1R)
├─ Average R-Multiple: +1.42
└─ Standard Deviation: 1.89R

Risk Analysis:
├─ Maximum Drawdown: -8.2% (-$8,200)
├─ Recovery Factor: 1.51 (Net Profit ÷ Max DD)
├─ Ulcer Index: 3.4% (Drawdown severity)
└─ Value at Risk (95%): -$1,240 per trade
```

### `trust metrics compare`
Compare performance across time periods:
```bash
trust metrics compare --period1 \"last-30-days\" --period2 \"previous-30-days\"

Output:
Performance Comparison
=====================
Metric                  Last 30d    Prev 30d    Change
─────────────────────────────────────────────────────
Total Return           +8.2%       +4.1%       ↗️ +4.1%
Sharpe Ratio           2.31        1.89        ↗️ +0.42
Profit Factor          2.85        2.12        ↗️ +0.73
Win Rate               68.5%       62.0%       ↗️ +6.5%
Max Drawdown           -3.1%       -5.8%       ↗️ +2.7%
Avg R-Multiple         +1.42       +1.18       ↗️ +0.24

Trend: 📈 IMPROVING (6/6 metrics better)
```

### Financial Metrics Calculators
Create in `core/src/calculators/metrics/`:

```rust
// profit_factor.rs
pub fn calculate_profit_factor(trades: &[ClosedTrade]) -> Option<Decimal> {
    let gross_profit = trades.iter()
        .filter(|t| t.pnl > Decimal::ZERO)
        .map(|t| t.pnl)
        .sum::<Decimal>();
    
    let gross_loss = trades.iter()
        .filter(|t| t.pnl < Decimal::ZERO)
        .map(|t| t.pnl.abs())
        .sum::<Decimal>();
        
    if gross_loss > Decimal::ZERO {
        Some(gross_profit / gross_loss)
    } else {
        None
    }
}

// expectancy.rs  
pub fn calculate_expectancy(trades: &[ClosedTrade]) -> Decimal {
    if trades.is_empty() {
        return Decimal::ZERO;
    }
    
    let win_rate = calculate_win_rate(trades);
    let avg_win = calculate_average_win(trades);
    let avg_loss = calculate_average_loss(trades);
    
    (win_rate * avg_win) - ((1.0 - win_rate) * avg_loss)
}

// sharpe_ratio.rs
pub fn calculate_sharpe_ratio(
    returns: &[Decimal], 
    risk_free_rate: Decimal,
    period_days: u32
) -> Option<Decimal> {
    if returns.len() < 2 {
        return None;
    }
    
    let excess_returns: Vec<Decimal> = returns.iter()
        .map(|r| r - (risk_free_rate / Decimal::from(365) * Decimal::from(period_days)))
        .collect();
        
    let mean_excess = excess_returns.iter().sum::<Decimal>() / Decimal::from(excess_returns.len());
    let std_dev = calculate_standard_deviation(&excess_returns);
    
    if std_dev > Decimal::ZERO {
        Some(mean_excess / std_dev * Decimal::from(252).sqrt()) // Annualized
    } else {
        None
    }
}
```

### Configuration Management
Add configurable parameters:
```bash
trust config set risk-free-rate 0.05  # 5% annual
trust config set var-confidence 0.95   # 95% VaR
trust config show metrics             # Show all metric configs
```

### Export and Formatting Options
```bash
trust metrics advanced --format json > metrics.json
trust metrics advanced --format csv > metrics.csv
trust metrics compare --export excel --file performance.xlsx
```

### Historical Analysis
```bash
trust metrics history --metric sharpe-ratio --chart
trust metrics rolling --window 30 --metric profit-factor
```

## Acceptance Criteria
- [ ] All advanced metrics calculate correctly with edge cases
- [ ] `trust metrics advanced` shows comprehensive overview
- [ ] `trust metrics compare` enables period comparisons  
- [ ] Risk-free rate and other parameters are configurable
- [ ] Handles insufficient data gracefully (< 2 trades, etc.)
- [ ] Export formats (JSON, CSV) work correctly
- [ ] Statistical calculations are mathematically sound
- [ ] Unit tests for all metric calculations
- [ ] Integration tests for CLI commands
- [ ] Performance is reasonable with 1000+ trades

## Implementation Notes
- Use proper statistical libraries for complex calculations
- Ensure all ratios handle division by zero gracefully
- Consider caching expensive calculations for large datasets
- Make time period adjustments (daily vs annual) configurable
- Add validation for minimum data requirements per metric
- Integration with Phase 1 transaction data and Phase 2 grading data
